src/vs/editor/common/modes.ts

```diff
@@ -392,7 +392,7 @@ export interface ReferenceProvider {
 	/**
 	 * Provide a set of project-wide references for the given position and document.
 	 */
-	provideReferences(model: editorCommon.IReadOnlyModel, position: Position, context: ReferenceContext, token: CancellationToken): Location[] | Thenable<Location[]>;
+	provideReferences(model: editorCommon.IReadOnlyModel, position: Position, context: ReferenceContext, progress: (value: Location) => void, token: CancellationToken): Location[] | Thenable<Location[]>;
```

Comment:
@sqs MS was against adding progress to the Promise type and I agree with them, it opens up all kinds of questions about chaining